src/Units/PiperunnerModel.java

```diff
@@ -0,0 +1,27 @@
+package Units;
+
+import Engine.UnitActionType;
+import Terrain.TerrainType;
+import Units.MoveTypes.MoveType;
+import Units.Weapons.PipeGun;
+import Units.Weapons.WeaponModel;
+
+public class PiperunnerModel extends UnitModel
+{
+  private static final int UNIT_COST = 20000;
+  private static final int MAX_FUEL = 99;
+  private static final int IDLE_FUEL_BURN = 0;
+  private static final int VISION_RANGE = 4;
+  private static final int MOVE_POWER = 9;
+
+  private static final MoveType moveType = new MoveType();
+  private static final UnitActionType[] actions = UnitActionType.COMBAT_VEHICLE_ACTIONS;
+  private static final WeaponModel[] weapons = { new PipeGun() };
+
+  public PiperunnerModel()
```

Comment:
Still not a fan of Piperunners. If we are gonna add them, we might wanna rethink the "pipe" sprite we have too.

Reply:
Hah, perhaps.
We could add some train-tracks terrain and let them move on those. ~~and maybe store units like the Pain Train from SFW can~~

It occurs to me that Piperunners also conflict with the idea of making pipeseams just a "unit" you place on the map.
Maybe locations can/should have a "this unit" sort of variable that they pass out instead of null when calling `getResident()`? Except *that* might mean you can't actually stand on them... This probably calls for more functions with more clear and specific use cases.

Reply:
I don't think I'd make pipe seams a unit. I mean, we haven't addressed destructible terrain yet, but that doesn't seem like the first avenue I'd go down.

Reply:
Mm. Wargroove made destructible terrain a unit, and it's the easiest naive approach to the problem.
Though yeah, destructible terrain being its own thing is probably more desirable (especially if we want to do stuff like the fancy devices)
That'd require adding a whole new "system" and modifying our targeting logic, so I'm not super enthused... but honestly it'll probably be less complicated than I think.

Reply:
I like the idea of trains at some point, maybe. I was considering making pipes an electric/plasma fence, which would look cool and be appropriate for the job, but y'know, time.

In general though, if you want something to fill a new kind of role, it should be a new kind of thing. The League of Legends folks have an in-joke whenever unexpected behavior occurs that "it's coded as minions" - things like player-generated walls are literally just a row of minions with different graphics, and don't always act as much like a wall as you'd want.